lib/mapnik_backend.js

```diff
@@ -220,51 +227,91 @@ MapnikSource.prototype._createPool = function(xml) {
     if (this._pool) return;
 
     const factory = {
-        create: (callback) => {
-            try {
-                const map = new mapnik.Map(this._uri.query.tileSize, this._uri.query.tileSize);
-                map.bufferSize = this._uri.query.bufferSize;
-                const map_opts = {strict: this._uri.strict || false, base: this._base + '/'};
-                return map.fromString(xml, map_opts, callback);
-            } catch (err) {
-                return callback(err);
-            }
+        // This function should never reject ¯\_(ツ)_/¯
+        // see https://github.com/coopernurse/node-pool/issues/175
+        // see https://github.com/coopernurse/node-pool/issues/183
+        create: () => {
+            return new Promise((resolve) => {
+                try {
+                    const { tileSize, bufferSize } = this._uri.query;
+                    const map = new mapnik.Map(tileSize, tileSize);
+
+                    map.bufferSize = bufferSize;
+
+                    const map_opts = {strict: this._uri.strict || false, base: this._base + '/'};
+                    map.fromString(xml, map_opts, (err, map) => {
+                        if (err) {
+                            return resolve(err)
+                        }
+
+                        return resolve(map);
+                    });
+                } catch (err) {
+                    return resolve(err);
+                }
+            });
         },
         destroy : (map) => {
-            // see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Delete_in_strict_mode
-            map = null;
-        },
-        max: this._uri.query.poolSize || process.env.UV_THREADPOOL_SIZE || require('os').cpus().length
+            return new Promise((resolve) => {
+                // see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Delete_in_strict_mode
+                map = null;
+
+                return resolve();
+            });
+        }
     };
 
-    this._pool = Pool(factory);
+    const options = {
+        max: this._uri.query.poolSize || cpusNumber,
+        maxWaitingClients: this._uri.query.poolMaxWaitingClients ? this._uri.query.poolMaxWaitingClients : 32
```

Comment:
I think the default value should be a multiplier of `max`

Reply:
Let's see how these values work first.